File: software/examples/csharp/ExampleSimple.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your TNG DI8

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		TNGDI8 di8 = new TNGDI8(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get current values
		bool[] values = di8.GetValues();

		Console.WriteLine("Channel 0: " + values[0]);
		Console.WriteLine("Channel 1: " + values[1]);
		Console.WriteLine("Channel 2: " + values[2]);
		Console.WriteLine("Channel 3: " + values[3]);
		Console.WriteLine("Channel 4: " + values[4]);
		Console.WriteLine("Channel 5: " + values[5]);
		Console.WriteLine("Channel 6: " + values[6]);
		Console.WriteLine("Channel 7: " + values[7]);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/rust/example_simple.rs
use std::{error::Error, io};

use tinkerforge::{di8_tng::*, ip_connection::IpConnection};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your TNG DI8.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let di8 = Di8Tng::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Get current values.
    let values = di8.get_values().recv()?;

    println!("Channel 0: {}", values[0]);
    println!("Channel 1: {}", values[1]);
    println!("Channel 2: {}", values[2]);
    println!("Channel 3: {}", values[3]);
    println!("Channel 4: {}", values[4]);
    println!("Channel 5: {}", values[5]);
    println!("Channel 6: {}", values[6]);
    println!("Channel 7: {}", values[7]);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
